### PR TITLE
feat: forbid throw new error

### LINKED ISSUE #20: No default thrown errors
We would like to 'permit' throwing of default errors like
`Throw new Error()`

Errors  manually thrown should allways be locally implemented and should contain a detailed message, status, etc...

### PR DESCRIPTION
**Pull request by vuki656**
Closes #20

### CODE REVIEW

**Review by andro-bosnjak — APPROVED** [review]

File: .github/workflows/default.yml [hunk 1]
```diff
@@ -21,7 +21,10 @@ jobs:
             - run: yarn install
               env:
                   NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
-            - run: yarn lint
+            # There is a weird bug in typescript eslint where if fails if you are linting an eslint
+            # plugin with a config that uses that same eslint plugin. We need the latest typescript eslint versions
```
Comment (mmilardic) [inline .github/workflows/default.yml]: Not sure what was the intention of the first sentence 👯, but I guess that we can't use ESlint to lint this plugin

File: src/rules/forbid-throw-new-error.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { createRule } from '../utils'
+
+const NAME = 'forbid-throw-new-error'
+
+const value = createRule({
+    create(context) {
+        return {
+            Identifier(node) {
+                if (node.name === "Error") {
```
Comment (mmilardic) [inline src/rules/forbid-throw-new-error.ts:9]: This completely forbids use of 'Error'?

Comment (vuki656) [reply src/rules/forbid-throw-new-error.ts:9]: Yup, pretty much a shotgun approach. Will test it on the backend before making it official ofc.

File: .github/workflows/default.yml (hunk 1 above)
Comment (vuki656) [reply .github/workflows/default.yml]: TLDR: A uses B, B uses A. Circular dep that I guess eslint plugin doesn't like. I should probably open an issue about that at some point.

**Review by mmilardic — APPROVED** [review]

**Comment by Spolja** [comment]
Should we merge this @vuki656  ?

**Pull request merged**